### ISSUE TITLE
Rialto <> Millau bridge issue

### ISSUE THREAD

**Issue by svyatonik** [paritytech/parity-bridges-common#2146]
This has happened over the weekend. Probably caused by #2129

```
bridges-bot
[[Alerting] Messages from Millau to Rialto are not being delivered](http://grafana.millau.brucke.link/d/relay-millau-to-rialto-messages/millau-to-rialto-message-sync-dashboard?tab=alert&viewPanel=10&orgId=1)
Millau Messages delivered to Rialto in last 5m: 0
[[Alerting] Messages from Rialto to Millau are not being delivered](http://grafana.millau.brucke.link/d/relay-rialto-to-millau-messages/rialto-to-millau-message-sync-dashboard?tab=alert&viewPanel=10&orgId=1)
Rialto Messages delivered to Millau in last 5m: 0
bridges-bot
[[Alerting] Rialto -> Millau messages are not detected by relay](http://grafana.millau.brucke.link/d/relay-rialto-to-millau-messages/rialto-to-millau-message-sync-dashboard?tab=alert&viewPanel=6&orgId=1)
Messages generated in last 5 minutes (Rialto -> Millau, 00000000): 0
[[Alerting] Millau -> Rialto messages are not detected by relay](http://grafana.millau.brucke.link/d/relay-millau-to-rialto-messages/millau-to-rialto-message-sync-dashboard?tab=alert&viewPanel=6&orgId=1)
Messages generated in last 5 minutes (Millau -> Rialto, 00000000): 0
```

**Comment by svyatonik** [comment]
Seems like some bug with new subscriptions. Last log lines:
```
[1;38;5;8m2023-05-24 05:02:19 +00[0m [38;5;10mINFO[0m [38;5;8mbridge[0m Prepared initialization data for Millau-headers bridge at RialtoParachain: InitializationData { header: Header { parent_hash: 0x621c7fc736e8b764455b6905f8a6cdd9436bff027eaa159211c47315726912d3e364e6fd94bb6243c39792b44292cfd96e6729d140aa99b07c3a4e5104a5f717, number: 3, state_root: 0xc4d9e65a84f0c5c75feca0dfb645601a506267c5a2fe13e8e9647100edc39747d860d044f1d56b287c723a305544eab9b46f765f4766ba8523096b4b01bf7a4d, extrinsics_root: 0xc5e7a3d94efcaedfe2f705654867618859909a9b720bbaf8a2cc4a17a93c248c37129d859d1915d6eb1680b29acd87481066cde001a4299c7bc7bd4ae34714a5, digest: Digest { logs: [DigestItem::PreRuntime([97, 117, 114, 97], [13, 239, 188, 16, 0, 0, 0, 0]), DigestItem::Consensus([66, 69, 69, 70], [3, 124, 195, 143, 16, 242, 149, 204, 232, 237, 180, 219, 157, 227, 54, 123, 236, 134, 251, 220, 119, 42, 157, 155, 112, 247, 11, 146, 49, 52, 68, 161, 92]), DigestItem::Seal([97, 117, 114, 97], [170, 138, 146, 95, 105, 48, 110, 216, 194, 227, 47, 119, 44, 224, 170, 50, 253, 123, 43, 118, 126, 41, 148, 149, 189, 152, 200, 205, 185, 105, 186, 51, 36, 12, 2, 170, 54, 2, 11, 199, 20, 188, 225, 33, 151, 58, 73, 135, 71, 66, 103, 153, 241, 224, 252, 244, 32, 168, 38, 168, 216, 160, 61, 131])] } }, authority_list: [(Public(88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee (5FA9nQDV...)), 1), (Public(d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae69 (5GoNkf6W...)), 1), (Public(439660b36c6c03afafca027b910b4fecf99801834c62a5e6006f27d978de234f (5DbKjhNL...)), 1), (Public(5e639b43e0052c47447dac87d6fd2b6ec50bdd4d0f614e4299c665249bbd09d9 (5ECTwv6c...)), 1), (Public(1dfe3e22cc0d45c70779c1095f7489a8ef3cf52d62fbd8c2fa38c9f1723502b5 (5Ck2miBf...)), 1)], set_id: 0, operating_mode: BasicOperatingMode::Normal }
[1;38;5;8m2023-05-24 05:02:19 +00[0m [38;5;10mINFO[0m [38;5;8mbridge[0m Initialize bridge call encoded as hex string: "0x02003701621c7fc736e8b764455b6905f8a6cdd9436bff027eaa159211c47315726912d3e364e6fd94bb6243c39792b44292cfd96e6729d140aa99b07c3a4e5104a5f7170cc4d9e65a84f0c5c75feca0dfb645601a506267c5a2fe13e8e9647100edc39747d860d044f1d56b287c723a305544eab9b46f765f4766ba8523096b4b01bf7a4dc5e7a3d94efcaedfe2f705654867618859909a9b720bbaf8a2cc4a17a93c248c37129d859d1915d6eb1680b29acd87481066cde001a4299c7bc7bd4ae34714a50c0661757261200defbc1000000000044245454684037cc38f10f295cce8edb4db9de3367bec86fbdc772a9d9b70f70b92313444a15c05617572610101aa8a925f69306ed8c2e32f772ce0aa32fd7b2b767e299495bd98c8cdb969ba33240c02aa36020bc714bce121973a498747426799f1e0fcf420a826a8d8a03d831488dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee0100000000000000d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae690100000000000000439660b36c6c03afafca027b910b4fecf99801834c62a5e6006f27d978de234f01000000000000005e639b43e0052c47447dac87d6fd2b6ec50bdd4d0f614e4299c665249bbd09d901000000000000001dfe3e22cc0d45c70779c1095f7489a8ef3cf52d62fbd8c2fa38c9f1723502b50100000000000000000000000000000000"
[1;38;5;8m2023-05-24 05:02:19 +00[0m [38;5;12mTRACE[0m [38;5;8mbridge[0m Sent transaction to RialtoParachain node: 0x3a1c45e54743b889da46b9e515f56f63a497090faf5d73a5c18931e3e7ab39fd
```

**Comment by svyatonik** [comment]
Reverted new client temporarily (#2153) to understand what is causing issues.

**Comment by svyatonik** [comment]
Restarted test deployments without new client. Current plan is to:
- run no-new-client version on test deployments for couple of days and see if the issue is there. If it is, then we have some deps-update related issue (misconfig probbaly). If there's no issue - then the new client is the cause;
- at the same time - run new-client and updated deps version locally and look for issues.

**Comment by svyatonik** [comment]
Looks like the cause of Rialto <> Millau bridge break is the new client (can't say the same regarding parachain registration):
```
[Rialto_to_Millau_MessageLane_00000000] [1;38;5;8m2023-05-24 16:40:03 +00[0m [38;5;14mDEBUG[0m [38;5;8mbridge[0m Target Rialto::ReceivingConfirmationsDelivery client has been asked for more Millau::ReceivingConfirmationsDelivery headers. Batch tx: no
[Rialto_to_Millau_MessageLane_00000000] [1;38;5;8m2023-05-24 16:40:03 +00[0m [38;5;14mDEBUG[0m [38;5;8mbridge[0m Received best nonces from Millau::MessagesDelivery: TargetClientNonces { latest_nonce: 4517, nonces_data: DeliveryRaceTargetNoncesData { confirmed_nonce: 4513, unrewarded_relayers: UnrewardedRelayersState { unrewarded_relayer_entries: 1, messages_in_oldest_entry: 4, total_messages: 4, last_delivered_nonce: 4517 } } }
[Rialto_to_Millau_MessageLane_00000000] [1;38;5;8m2023-05-24 16:40:03 +00[0m [38;5;14mDEBUG[0m [38;5;8mbridge[0m Asking Millau::MessagesDelivery about finalized message nonces
[Rialto_to_Millau_MessageLane_00000000] [1;38;5;8m2023-05-24 16:40:03 +00[0m [38;5;14mDEBUG[0m [38;5;8mbridge[0m Received finalized nonces from Millau::MessagesDelivery: TargetClientNonces { latest_nonce: 4517, nonces_data: DeliveryRaceTargetNoncesData { confirmed_nonce: 4513, unrewarded_relayers: UnrewardedRelayersState { unrewarded_relayer_entries: 1, messages_in_oldest_entry: 4, total_messages: 4, last_delivered_nonce: 4517 } } }
[1;38;5;8m2023-05-24 16:40:07 +00[0m [38;5;12mTRACE[0m [38;5;8mbridge-metrics[0m Refreshed process metrics: CPU=0.20149864256381989, memory=189372825600
[Millau_to_Rialto_MessageLane_00000000] [1;38;5;8m2023-05-24 16:40:07 +00[0m [38;5;12mTRACE[0m [38;5;8mbridge[0m Millau transaction 0x7d4f434635bea9a67ef6fd732c36c56a2d05446f6003f9eb39722b231facd39b21c42d4da60295c215cfe6a4d6ea8f245821db07726030605f4fe2ef272d87e4 has been finalized at block: 0xa4ebab66df633a773dae1617cf21a147275a3e84172d37e02217811e31580bfaa248df6ca4584483fe935926e3ecc22394d3de4ecdbbb0737efc881ea8dbb0bd
[1;38;5;8m2023-05-24 16:40:07 +00[0m [38;5;14mDEBUG[0m [38;5;8mbridge[0m Background task of transaction events subscription of Millau has stopped: client has stopped
[Millau_to_Rialto_MessageLane_00000000] [1;38;5;8m2023-05-24 16:40:08 +00[0m [38;5;14mDEBUG[0m [38;5;8mbridge[0m Asking Millau node about its state
[Rialto_to_Millau_MessageLane_00000000] [1;38;5;8m2023-05-24 16:40:08 +00[0m [38;5;14mDEBUG[0m [38;5;8mbridge[0m Asking Millau node about its state
```
And then - no new log lines for ~1h (apart from standalone metric updates).

**Comment by svyatonik** [comment]
fixed in #2164
